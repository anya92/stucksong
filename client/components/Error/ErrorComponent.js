import React from 'react';
import { string } from 'prop-types';
import { ErrorContainer } from '../../styles/info';

const ErrorComponent = ({ error }) => (
  <ErrorContainer>
    <h1>Error <span role="img" aria-label="sad-emoji">ðŸ˜ž</span></h1>
    <h2>Please try again later.</h2>
    <p>Additional info: {error}</p>
  </ErrorContainer>
);

ErrorComponent.propTypes = {
  error: string,
};

ErrorComponent.defaultProps = {
  error: '',
};

export default ErrorComponent;
