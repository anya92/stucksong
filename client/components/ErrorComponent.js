import React from 'react';
import { string } from 'prop-types';
import { ErrorContainer } from '../styles/error';

const ErrorComponent = ({ error }) => (
  <ErrorContainer>
    <h1>Error <span role="img" aria-label="sad-emoji">ðŸ˜ž</span></h1>
    <h2>Please try again later.</h2>
    <p>Additional info: {error}</p>
  </ErrorContainer>
);

ErrorComponent.propTypes = {
  error: string.isRequired,
};

export default ErrorComponent;
