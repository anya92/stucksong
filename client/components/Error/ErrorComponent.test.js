import React from 'react';
import { shallow } from 'enzyme';

import ErrorComponent from './ErrorComponent';

describe('ErrorComponent', () => {
  it('renders Error Component correctly', () => {
    const error = 'Request failed';
    const wrapper = shallow(<ErrorComponent error={error} />);
    expect(wrapper).toMatchSnapshot();
  });
});
