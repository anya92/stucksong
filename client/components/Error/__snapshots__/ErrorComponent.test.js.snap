// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent renders correctly 1`] = `
<glamorous(div)>
  <h1>
    Error 
    <span
      aria-label="sad-emoji"
      role="img"
    >
      ðŸ˜ž
    </span>
  </h1>
  <h2>
    Please try again later.
  </h2>
  <p>
    Additional info: 
    Request failed
  </p>
</glamorous(div)>
`;
